ui/tests/test_learningresources_views.py

```diff
@@ -237,3 +256,63 @@ def test_repo_course_filter(self):
             self.course.course_number)
         resp = self.client.get(self.repository_url + querystring, follow=True)
         self.assertTrue(resp.status_code == HTTP_OK)
+
+    def test_serve_media(self):
+        """Hit serve media"""
+        self.assertEqual(
+            settings.DEFAULT_FILE_STORAGE,
+            'django.core.files.storage.FileSystemStorage'
+        )
+        # upload a course
+        self.upload_test_file()
+        self.assertEqual(len(StaticAsset.objects.all()), 3)
+        # take the url of a static asset
+        static_asset_url = StaticAsset.objects.all()[0].asset.url
+        # hit the view
+        resp = self.client.get(static_asset_url)
+        self.assertEqual(resp.status_code, HTTP_OK)
+        self.assertEqual(
+            resp.get('Content-Disposition'),
+            'attachment; filename={}'.format(
+                os.path.basename(static_asset_url)
+            )
+        )
+        # only the user with right to see the repo can access the file
+        self.logout()
+        self.login(self.user_norepo.username)
+        resp = self.client.get(static_asset_url)
+        self.assertEqual(resp.status_code, UNAUTHORIZED)
+        # login back with the original user
+        self.logout()
+        self.login(self.user.username)
+        # hit the view with a nonexistent file
+        resp = self.client.get('/media/fsdfs2837hwdudnks/foo.txt')
+        self.assertEqual(resp.status_code, NOT_FOUND)
+        # change the default file storage to S3
+        with self.settings(
+            DEFAULT_FILE_STORAGE=('storages.backends'
+                                  '.s3boto.S3BotoStorage')
+        ):
+            # force the reload of the urls
+            try:
+                reload(ui.urls)  # pylint: disable=undefined-variable
+            # this is in case of python 3.4
```

Comment:
Can you write a brief comment above all this explaining why python 3.3 and 3.4 are raising exceptions here?